src/parser/xasl_generation.c

```diff
@@ -3119,6 +3224,29 @@ pt_null_xasl (PARSER_CONTEXT * parser, PT_NODE * tree, void *void_arg, int *cont
   return tree;
 }
 
+/*
+ * pt_is_spec_node () - return node with the same spec id
+ *   return:
+ *   parser(in):
+ *   tree(in):
+ *   void_arg(in):
+ *   continue_walk(in):
+ */
+static PT_NODE *
+pt_is_spec_node (PARSER_CONTEXT * parser, PT_NODE * tree, void *void_arg, int *continue_walk)
+{
+  if (pt_is_name_node (tree))
+    {
+      UINTPTR *spec_id = (UINTPTR *) void_arg;
+      if (tree->info.name.spec_id == spec_id[0])
+	{
+	  *continue_walk = PT_STOP_WALK;
+	  spec_id[1] = 1;
```

Comment:
how is spec_id guaranteed that it has enough memory allocated?

Reply:
pt_is_spec_node() is invoked only from pt_split_hash_attrs().
```
              UINTPTR spec_id[2], spec_id2[2];
	      spec_id[0] = spec_id2[0] = table_info->spec_id;
	      spec_id[1] = spec_id2[1] = 0;
	      parser_walk_tree (parser, arg1, pt_is_spec_node, &spec_id, NULL, NULL);
	      parser_walk_tree (parser, arg2, pt_is_spec_node, &spec_id2, NULL, NULL);
```